src/modules/auth/auth.service.ts

```diff
@@ -82,13 +89,54 @@ export class AuthService {
     const accessToken = await this.generateAuthToken(payload, 'access');
     const refreshToken = await this.generateAuthToken(payload, 'refresh');
 
+    const refreshTokenEntity = this.createRefreshTokenEntity(refreshToken);
+    await this.refreshTokenRepository.insert(refreshTokenEntity);
+
     return {
       message: 'User logged in successfully',
       accessToken,
       refreshToken,
     };
   }
 
+  async issueNewTokens(refreshToken: string) {
+    const decodedRefreshToken =
+      this.jwtService.decode<AuthTokenPayload>(refreshToken);
+
```

Comment:
Valdiate the token instead of decoding here. Throw an exception if it fails. Decode does not verify the signature and expiry status of token.

Reply:
> Valdiate the token instead of decoding here. Throw an exception if it fails. Decode does not verify the signature and expiry status of token.

Done